lib/modules/apostrophe-browser-utils/public/js/lean.js

```diff
@@ -87,11 +87,46 @@
   // Implementation detail of `apos.utils.post` and `apos.utils.get`
   function monitor(xmlhttp, callback) {
     xmlhttp.addEventListener("load", function() {
+      // Grab the response header;
+      var responseHeader = this.getResponseHeader("Content-Type");
+      // Trim the response so we can accurately test;
+      var trimResponse = this.responseText.trim();
       var data;
-      try {
-        data = JSON.parse(this.responseText);
-      } catch (e) {
-        return callback(e);
+      // If the response header is `application/json`, immediately try
+      // to JSON.parse(), and if it fails, report the error.
+      if (responseHeader === 'application/json; charset=utf-8') {
+        try {
+          data = JSON.parse(this.responseText);
+        } catch (e) {
+          return callback(e);
+        }
+      // If there is no explicit content-type header,
+      // test the trimmed response against our isJson() method.
+      // If it returns `true`, try to JSON.parse() it.
+      // And if it fails, try to JSON.stringify() it, and if that fails,
+      // report an error.
+      } else {
+        if (isJSON(trimResponse, callback)) {
+          try {
+            data = JSON.parse(this.responseText);
+          } catch (e) {
+            try {
+              data = JSON.stringify(this.responseText);
```

Comment:
wat why would you stringify it? That would turn:

OK

Into (as a string): `"OK"`

at this point we just want to give up and treat data as a string, which it already is.